content/03-reference/01-tools-and-interfaces/02-prisma-client/08-pagination/index.mdx

```diff
@@ -0,0 +1,213 @@
+
+> **Note**: Since this is the first query, there is no cursor to pass in.
+
+```ts line-number
+const firstQueryResults = prisma.post.findMany({
+  take: 4,
+  where: {
+    title: {
+      contains: 'Prisma' /* Optional filter */,
+    },
+  },
+  orderBy: {
+    id: 'asc',
+  },
+})
+
+// Bookmark your location in the result set - in this
+// case, the ID of the last post in the list of 4.
+
+|const lastPostInResults = firstQueryResults[3] // Remember: zero-based index! :)
+|const myCursor = lastPostInResults.id // Example: 29
+```
+
+The following diagram shows the IDs of the first 4 results - or page 1. The cursor for the next query is **29**:
+
+![](pagination/cursor-1.png)
+
+The second query returns the first 4 `Post` records that contain the word `"Prisma"` **after the supplied cursor** (in other words - IDs that are larger than **29**):
+
+```ts line-number
+const secondQueryResults = prisma.post.findMany({
+  take: 4,
+  skip: 1, // Skip the cursor
+|  cursor: {
+|    id: myCursor,
+|  },
+  where: {
+    title: {
+      contains: 'Prisma' /* Optional filter */,
+    },
+  },
+  orderBy: {
+    id: 'asc',
+  },
+})
+
+const lastPostInResults = secondQueryResults[3] // Remember: zero-based index! :)
+const myCursor = lastPostInResults.id // Example: 52
+```
+
+The following diagram shows the first 4 `Post` records **after** the record with ID **29**. In this example, the new cursor is **52**:
+
+![](pagination/cursor-2.png)
+
+### FAQ
+
+#### Do I always have to skip: 1?
+
+If you do not `skip: 1`, your result set will include your previous cursor. The first query returns four results and the cursor is **29**:
+
+![](pagination/cursor-1.png)
+
+Without `skip: 1`, the second query returns 4 results after (and _including_) the cursor:
+
+![](pagination/cursor-3.png)
+
+If you `skip: 1`, the cursor is not included:
+
+![](pagination/cursor-2.png)
+
+You can choose to `skip: 1` or not depending on the pagination behavior that you want.
+
+#### Can I guess the value of the cursor?
+
+If you guess the value of the next cursor, you will page to an unknown location in your result set. Although IDs are sequential, you cannot predict the rate of increment (`2`, `20`, `32` is more likely than `1`, `2` `3`, particularly in a filtered result set).
+
+#### Does cursor-based pagination use the concept of a cursor in the underlying database?
+
+No, cursor pagination does not [use cursors in the underlying database](https://www.postgresql.org/docs/9.2/plpgsql-cursors.html).
+
+### ✔ Pros of cursor-based pagination
+
+- Cursor-based pagination **scales**. The underlying SQL does not use `OFFSET`, but instead queries all `Post` records with an ID greater than X.
+
+### ✘ Cons of cursor-based pagination
+
+- You must sort by your cursor, which has to be a unique, sequential column.
+- You cannot jump to a specific page number - you must traverse pages 1 - 20 in order to know what the cursor should be for the query that returns page 21.
```

Comment:
Couldn't I also do a query with LIMIT 200 (in example with page size 10) and get the ID of the last entry to use as a cursor? Or is that meant by "traverse"?

Reply:
Updated this to say you cannot use *only* a cursor to jump to page X, because you don't know what the cursor is going to be. Removed the word 'traverse'.